LocationManager/INTULocationManager/INTULocationManager.m

```diff
@@ -692,6 +758,110 @@ - (INTULocationAccuracy)achievedAccuracyForLocation:(CLLocation *)location
     }
 }
 
+#pragma mark Internal heading methods
+
+/**
+ Adds the given heading request to the array of requests and starts heading updates.
+ */
+- (void)addHeadingRequest:(INTUHeadingRequest *)headingRequest
+{
+    NSAssert(headingRequest, @"Must pass in a non-nil heading request.");
+
+    // If heading services are not available, just return
+    if ([INTULocationManager headingServicesState] == INTUHeadingServicesStateUnavailable) {
+        if (headingRequest.block) {
+            headingRequest.block(nil, INTUHeadingStatusUnavailable);
```

Comment:
Calling the block synchronously like this is problematic because it means the block will execute before the initial `subscribeToHeadingUpdatesWithBlock:` call returns, which means that the client doesn't know or have the ID for the request (yet the code in the block may need to use that ID to know which request it corresponds to). At the very least, this needs to dispatch_async like we do for this code path with location requests.